File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/rust_solutions/2729-check-if-the-number-is-fascinating.rs
impl Solution {
    pub fn is_fascinating(n: i32) -> bool {
        let concatenated = format!("{}{}{}", n, 2 * n, 3 * n);
        if concatenated.len() != 9 {
            return false;
        }
        let mut digits = [false; 10];
        for c in concatenated.chars() {
            let digit = c.to_digit(10).unwrap() as usize;
            if digit == 0 || digits[digit] {
                return false;
            }
            digits[digit] = true;
        }
        true
    }
}

File: leetcode/AIcoding/3.DeepSeek-v3/c_solutions/2729-check-if-the-number-is-fascinating.c
bool isFascinating(int n) {
    int n2 = n * 2;
    int n3 = n * 3;

    int digits[10] = {0};

    while (n > 0) {
        digits[n % 10]++;
        n /= 10;
    }

    while (n2 > 0) {
        digits[n2 % 10]++;
        n2 /= 10;
    }

    while (n3 > 0) {
        digits[n3 % 10]++;
        n3 /= 10;
    }

    if (digits[0] > 0) return false;

    for (int i = 1; i <= 9; i++) {
        if (digits[i] != 1) return false;
    }

    return true;
}